src/Config.Binder/ConfigurationBinder.cs

```diff
@@ -179,6 +180,14 @@ private static void BindNonScalar(this IConfiguration configuration, object inst
         {
             if (instance != null)
             {
+                foreach(var requiredProperty in options.RequiredProperties)
+                {
+                    if (configuration.GetSection(requiredProperty)?.Value == null)
+                    {
+                        throw new SerializationException(Resources.FormatError_FailedBindingRequiredProperty(requiredProperty));
```

Comment:
I don't think a serialization exception is appropriate here, maybe InvalidOperationException?